lib/integrations/awsoidc/deployservice.go

```diff
@@ -0,0 +1,671 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awsoidc
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/ecs"
+	ecsTypes "github.com/aws/aws-sdk-go-v2/service/ecs/types"
+	"github.com/gravitational/trace"
+	"golang.org/x/exp/slices"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/utils/retryutils"
+)
+
+var (
+	// launchTypeFargateString is the FARGATE LaunchType converted into a string.
+	launchTypeFargateString = string(ecsTypes.LaunchTypeFargate)
+	// requiredCapacityProviders contains the FARGATE type which is required to deploy a Teleport Service.
+	requiredCapacityProviders = []string{launchTypeFargateString}
+
+	// Ensure Cpu and Memory use one of the allowed combinations:
+	// https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html
+	taskCPU = "512"
```

Comment:
That's half a vCPU according to that link above, sounds like too little? Can we request at least 2 vCPUs?

Reply:
Sure 👍 
I'll update it to 2 vCPU and 4GB (min possible for that vCPU value)

There's also a price tradeoff to consider
0.512 vCPU / 1GB = ~17$/month
2 vCPU / 4GB = ~71$/month

(with 2vCPU you must have between 4GB and 16GB of RAM)